ui/js/signup.js

```diff
@@ -0,0 +1,358 @@
+/*!
+ * Copyright 2015 Apereo Foundation (AF) Licensed under the
+ * Educational Community License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License. You may
+ * obtain a copy of the License at
+ *
+ *     http://opensource.org/licenses/ECL-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an "AS IS"
+ * BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+require(['jquery', 'underscore', 'oae.core'], function($, _, oae) {
+
+    var DEFAULT_SIGN_UP_REDIRECT_URL = '/me';
+
+    var signUpRedirectUrl = null;
+    var invitationInfo = null;
+    var pageTitle = null;
+    var authStrategyInfo = null;
+    var recaptchaEnabled = null;
```

Comment:
This format felt clearer to me because:

1. This way it's clear what stateful variables are available in the module scope
2. Many of the variables have non-trivial logic to initialize them that I wanted to live in functions rather than module scope
3. Many of the variables are pre-requisites to the others, so the variable declarations and logic would be interweaved in the module scope